modules/core/shared/src/main/scala/crystal/Pot.scala

```diff
@@ -66,6 +66,14 @@ sealed trait Pot[+A] {
     }
 
   def toOptionEither: Option[Either[Throwable, A]] = toOptionTry.map(_.toEither)
+
+  def filter(f: A => Boolean): Pot[A] =
+    this match {
+      case Pot.Ready(a) if !f(a) => Pot.pending
+      case _                     => this
+    }
+
+  def filterNot(f: A => Boolean): Pot[A] = filter(a => !f(a))
```

Comment:
So, if you filter something out, it results in a Pot that never leaves the pending state? What is the use case for this?

Reply:
It plays well with the `useEffectWhenDepsReady` hook. I'm using it to run an effect when `odbStatusPot.filter(_ === PersistentClientStatus.Initialized)`.